.github/workflows/main.yaml

```diff
@@ -15,11 +15,52 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
       - name: Setup Java
         uses: actions/setup-java@v2
         with:
           java-version: '8'
           distribution: 'adopt'
           cache: 'maven'
+      - uses: ./.github/actions/setup-git
+        with:
+          github_token:   ${{ secrets.GITHUB_TOKEN }}
+          git_user_email: ${{ secrets.RELEASE_GH_EMAIL }}
+          git_username:   ${{ secrets.RELEASE_GH_USERNAME }}
+      - name: Set version (master only)
+        if: github.ref == 'refs/heads/master'
+        run: mvn --batch-mode release-candidate:updateVersion
       - name: Build with Maven
         run: mvn --batch-mode --update-snapshots verify
+      - name: Read $version (master only)
+        if: github.ref == 'refs/heads/master'
+        run: cat build-monitor-plugin/target/classes/build-monitor.properties >> $GITHUB_ENV
+      - uses: actions/upload-artifact@v2
+        with:
+          retention-days: 3
+          name: artifacts
+          path: |
+            build-monitor-plugin/target/*.hpi
+            pom.xml
+            build-monitor-plugin/pom.xml
+            build-monitor-acceptance/pom.xml
+            build-monitor-acceptance/build-monitor-acceptance-base/pom.xml
+            build-monitor-acceptance/build-monitor-acceptance-latest/pom.xml
+            build-monitor-acceptance/build-monitor-acceptance-base/target/failsafe-reports/*-output.txt
+            build-monitor-acceptance/build-monitor-acceptance-latest/target/failsafe-reports/*-output.txt
+      - name: Create Release Branch (master only)
+        if: github.ref == 'refs/heads/master'
+        run: |
+          git checkout -b "release-${version}"
+          git commit -a -m "Release candidate v$version"
+          git push origin "release-$version"
+#      - name: Publish (master only)
```

Comment:
FTR this is possible in the jenkinsci org without needing to use your own credentials

https://www.jenkins.io/jep/229
https://www.jenkins.io/doc/developer/publishing/releasing-cd/

Reply:
Thanks, I'll take a look.